assisted_service_mcp/src/tools/network_tools.py

```diff
@@ -0,0 +1,204 @@
+    return "YAML is valid"
+
+
+@track_tool_usage()
+async def generate_nmstate_yaml(
+    _get_access_token_func: Callable[[], str],
+    params: Annotated[
+        NMStateTemplateParams,
+        Field(
+            description="Structured network configuration parameters including interface name, IP addresses (IPv4/IPv6), DNS servers, gateway, and routes. Use NMStateTemplateParams schema."
+        ),
+    ],
+) -> str:
+    """Generate NMState YAML from structured network parameters.
+
+    Creates NMState YAML configuration from structured parameters rather than writing YAML
+    manually. Always use this to generate initial YAML from user requirements, then validate
+    and optionally tweak the result. Do not generate nmstate yaml from scratch without calling
+    this tool.
+
+    Prerequisites:
+        - Network information from user (interface, IPs, gateway, DNS)
+
+    Related tools:
+        - validate_nmstate_yaml - Validate the generated YAML
+        - alter_static_network_config_nmstate_for_host - Apply generated YAML to hosts
+        - list_static_network_config - View applied configurations
+
+    Returns:
+        str: Generated nmstate YAML or error message.
+    """
+    log.info("Generate nmstate yaml with params: %s", params.model_dump_json(indent=2))
+    try:
+        generated = generate_nmstate_from_template(params)
+        log.debug("Generated yaml: %s", generated)
+        return generated
+    except TemplateError as e:
+        log.error("Failed to render nmstate template", exc_info=e)
+        return "ERROR: Failed to generate nmstate yaml"
+    except Exception as e:
+        log.error("Exception generating nmstate yaml", exc_info=e)
+        return "ERROR: Unknown error"
+
+
+@track_tool_usage()
+async def alter_static_network_config_nmstate_for_host(
+    get_access_token_func: Callable[[], str],
+    cluster_id: Annotated[
+        str,
+        Field(description="The unique identifier of the cluster to configure."),
+    ],
+    index: Annotated[
+        int | None,
+        Field(
+            description="The position of the host in the static network configuration list. Use None to append a new host configuration. Use 0, 1, 2, etc. to replace or delete an existing host configuration."
+        ),
+    ],
+    new_nmstate_yaml: Annotated[
+        str | None,
+        Field(
+            description="The validated NMState YAML to apply. Use None to delete the configuration at the specified index. Provide YAML to add or update a configuration."
+        ),
+    ],
+) -> str:
+    r"""Add, replace, or delete static network configuration for a host.
+
+    Manages static network configurations for cluster hosts. To add a new host config, use
+    index=None and provide YAML. To update an existing host config, provide the index and
+    new YAML. To remove a host config, provide the index and set YAML=None. Each
+    configuration corresponds to one host in the order they boot from the ISO.
```

Comment:
Is this really the ordering? That's interesting, but also is it useful? Is it something the model can check given context?